specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,370 @@
+  @doc("Message parameter type is text.")
+  kind: MessageTemplateValueType.text;
+
+  @doc("The text value.")
+  text: string;
+}
+
+@doc("""
+The message template's media value information.
+Could be an image, document or
+video.
+""")
+model MessageTemplateMedia {
+  @projectedName("csharp", "uri")
+  @doc("The (public) URL of the media.")
+  url: url;
+
+  @doc("The [optional] caption of the media object.")
+  caption?: string;
+
+  @doc("The [optional] filename of the media file.")
+  fileName?: string;
+}
+
+@doc("The message template's image value information.")
+model MessageTemplateImage extends MessageTemplateValue {
+  @doc("Message parameter type is image.")
+  kind: MessageTemplateValueType.image;
+
+  ...MessageTemplateMedia;
+}
+
+@doc("The message template's document value information.")
+model MessageTemplateDocument extends MessageTemplateValue {
+  @doc("Message parameter type is document.")
+  kind: MessageTemplateValueType.document;
+
+  ...MessageTemplateMedia;
+}
+
+@doc("The message template's video value information.")
+model MessageTemplateVideo extends MessageTemplateValue {
+  @doc("Message parameter type is video.")
+  kind: MessageTemplateValueType.video;
+
+  ...MessageTemplateMedia;
+}
+
+@doc("The message template's location value information.")
+model MessageTemplateLocation extends MessageTemplateValue {
+  @doc("Message parameter type is location.")
+  kind: MessageTemplateValueType.location;
+
+  @doc("The [Optional] name of the location.")
+  locationName?: string;
+
+  @doc("The [Optional] address of the location.")
+  address?: string;
+
+  @doc("The latitude of the location.")
+  latitude: float64;
+
+  @doc("The longitude of the location.")
+  longitude: float64;
+}
+
+@doc("The message template's quick action value information.")
+model MessageTemplateQuickAction extends MessageTemplateValue {
+  @doc("Message parameter type is quick action.")
+  kind: MessageTemplateValueType.quickAction;
+
+  @doc("The [Optional] quick action text")
+  text?: string;
+
+  @doc("The [Optional] quick action payload")
+  payload?: string;
+}
+
+@discriminator("kind")
+@doc("The binding object to link values to the template specific locations")
+model MessageTemplateBindings {
+  @doc("The type discriminator describing a template bindings type.")
+  kind: MessageTemplateBindingsType;
+}
+
+@doc("The template bindings for WhatsApp")
+model WhatsAppMessageTemplateBindings extends MessageTemplateBindings {
+  @doc("MessageTemplateBindings is whatsApp")
+  kind: MessageTemplateBindingsType.whatsApp;
+
+  @doc("The header template bindings")
+  header?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The body template bindings")
+  body?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The footer template bindings")
+  footer?: WhatsAppMessageTemplateBindingsComponent[];
+
+  @doc("The button template bindings")
+  buttons?: WhatsAppMessageTemplateBindingsButton[];
```

Comment:
nit, all of them is an array, why only `buttons` got plural?

Reply:
Discussed in the review meeting.